library/src/scala/internal/TupledFunction.scala

```diff
@@ -0,0 +1,160 @@
+package scala
+package internal
+
```

Comment:
You need to keep an `object TupledFunction` in scala.internal to preserve compatibility with the reference compiler.